### ISSUE TITLE
(HAL-28) ATTESTERS ARE NOT INCENTIVIZED TO TRIGGER ACCEPTBLOCKCOMMITMENT

### ISSUE THREAD

**Issue by SA124** [movementlabsxyz/movement#582]
(HAL-28) ATTESTERS ARE NOT INCENTIVIZED TO TRIGGER ACCEPTBLOCKCOMMITMENT

Auditor: Halborn
Severity: Medium

Description
Misaligned incentives for attesters block commitment can potentially leading to suboptimal network behavior, increased latency, and unfair distribution of costs and rewards.

The vulnerable logic is primarily in the tickOnBlockHeight function, which can trigger epoch rollovers:

![Screenshot 2024-09-12 at 11 27 16 AM](https://github.com/user-attachments/assets/4d9d969f-dded-43c9-9c84-ea32521cfac2)

BVSS
[AO:A/AC:L/AX:L/R:N/S:U/C:N/A:M/I:N/D:N/Y:M (6.3)](https://www.halborn.com/bvss?q=AO%3AA%2FAC%3AL%2FAX%3AL%2FR%3AN%2FS%3AU%2FC%3AN%2FA%3AM%2FI%3AN%2FD%3AN%2FY%3AM)

![Screenshot 2024-09-12 at 11 27 56 AM](https://github.com/user-attachments/assets/7a7f10f8-56e0-4402-b52e-d6a1a58ff0b5)

Recommendation
It is recommended to remove the burden of being the last to commit by correctly aligning rewards, without incentivizing attesters to be the last to commit. As an example, this could be done outside the contract, by creating a grant program for reimbursing attesters.

Trying to hide the state of the current validation may not be possible for attesters monitoring transactions and keeping track of the current state of the MCR.

**Comment by l-monninger** [comment]
Yep, apart from basic rewarding, this is a well-known issue that is a subject of research. See [MD-4](https://github.com/movementlabsxyz/MIP/pull/4)
